### ISSUE TITLE
No notifications, but the test message is okay

### ISSUE THREAD

**Issue by Yury-MonZon** [OctoPrint/OctoPrint-Pushbullet#20]
Test message works fine from the settings. But I'm not getting any other pushbullet notifications.

Probably, something in octoprint updates got awry, before it was working prefectly fine.

**Comment by foosel** [comment]
I cannot reproduce this, it works just fine over here against 1.3.6.

Please provide octoprint.log

**Comment by Yury-MonZon** [comment]
Here is it https://pastebin.com/QjUJc7LL
Thanks for your help!

**Comment by foosel** [comment]
```
ConnectionError: HTTPSConnectionPool(host='api.pushbullet.com', port=443): Max retries exceeded with url: /v2/pushes (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0xaf880c70>: Failed to establish a new connection: [Errno -2] Name or service not known',))
```

Looks like it wasn't able to resolve `api.pushbullet.com` when it was trying to send a message. Intermittent connection issues?

**Comment by Yury-MonZon** [comment]
Could be it. That wifi is kind of sluggish. But before it used to work fine. Any way to ease the pain?)

**Comment by foosel** [comment]
Not really, it can't send messages if it can't reach the service it's supposed to send message to.

**Comment by foosel** [comment]
Closing since it has apparently resolved itself.

**Comment by przemo** [comment]
I have the same problem. It worked at the first print, but after rebooting the server I get the same error. Is it possible that Pushbullet plugin tries to connect before Octoprint server is connected to the wifi? I do not think wifi is a problem - signal is good and no traffic.
```
2020-01-16 07:08:50,219 - octoprint.plugins.octobullet - ERROR - Error while instantiating PushBullet
Traceback (most recent call last):
  File "/home/pi/oprint/local/lib/python2.7/site-packages/octoprint_octobullet/__init__.py", line 394, in _create_sender
    bullet = pushbullet.PushBullet(token)
  File "/home/pi/oprint/local/lib/python2.7/site-packages/pushbullet/pushbullet.py", line 43, in __init__
    self.refresh()
  File "/home/pi/oprint/local/lib/python2.7/site-packages/pushbullet/pushbullet.py", line 381, in refresh
    self._load_devices()
  File "/home/pi/oprint/local/lib/python2.7/site-packages/pushbullet/pushbullet.py", line 78, in _load_devices
    resp_dict = self._get_data(self.DEVICES_URL)
  File "/home/pi/oprint/local/lib/python2.7/site-packages/pushbullet/pushbullet.py", line 64, in _get_data
    resp = self._session.get(url)
  File "/home/pi/oprint/local/lib/python2.7/site-packages/requests/sessions.py", line 546, in get
    return self.request('GET', url, **kwargs)
  File "/home/pi/oprint/local/lib/python2.7/site-packages/requests/sessions.py", line 533, in request
    resp = self.send(prep, **send_kwargs)
  File "/home/pi/oprint/local/lib/python2.7/site-packages/requests/sessions.py", line 646, in send
    r = adapter.send(request, **kwargs)
  File "/home/pi/oprint/local/lib/python2.7/site-packages/requests/adapters.py", line 516, in send
    raise ConnectionError(e, request=request)
ConnectionError: HTTPSConnectionPool(host='api.pushbullet.com', port=443): Max retries exceeded with url: /v2/devices (Caused by NewConnectionError('<urllib3.connection.VerifiedHTTPSConnection object at 0xaaa80dd0>: Failed to establish a new connection: [Errno -3] Temporary failure in name resolution',))
```

**Comment by JelledFro** [comment]
I also have this problem. The test message goes through, but I don't get a notification when a print is finished. My wifi connectivity seem perfectly fine for everything else and the pi haven't gone offline.

**Comment by mika1337** [comment]
I also have the same issue since a recent OctoPi the update.
I have the following logs:
`2020-04-09 07:00:50,768 - octoprint.plugin - ERROR - Error while calling plugin octobullet
Traceback (most recent call last):
  File "/home/pi/oprint/local/lib/python2.7/site-packages/octoprint/plugin/__init__.py", line 224, in call_plugin
    result = getattr(plugin, method)(*args, **kwargs)
  File "/home/pi/oprint/local/lib/python2.7/site-packages/octoprint_octobullet/__init__.py", line 241, in on_event
    path = os.path.basename(payload["file"])
KeyError: 'file' `

I tried to remove `{file}` from the message sent, but still have the issue.

**Comment by mika1337** [comment]
I’ve done some more tests and the payload given to the plugin is as follows :
`
{'origin': u'local', 'name': u'CE3_F9_Body_8_Mid.gcode', u'time': 14271.391323297, u'owner': 'octo', 'path': u'CE3_F9_Body_8_Mid.gcode', 'size': 13331124L}
`
As you could expect, `file` entry is missing from the payload and seemed to be replaced with `path`.

So a quick fix would be to replace:
`path = os.path.basename(payload["file"])`
with:
`path = os.path.basename(payload["path"]) `

**Comment by nlarson1981** [comment]
I recently encountered this issue as well after upgrading my Pi and a fresh OctoPrint install.  This plugin hasn't been updated in some time and is currently up for adoption.  Also, this plugin is not supported by Python 3 and Python 2 is pending EOL, so I found an alternative that works with pushbullet called Octoslack: https://github.com/fraschetti/Octoslack. Lots of customization options and works with multiple messaging platforms.  I recommend this as an alternative.
